File: src/app/services/role-service/role.service.ts
import { Injectable } from '@angular/core';
import {HttpBaseService} from "../http-base-service/http-base.service";
import {environment} from "../../../environments/environment.development";
import {APIResponseModel} from "../../model/interface";
import {Observable} from "rxjs";
import {Role} from "../../model/class";

@Injectable({
  providedIn: 'root'
})
export class RoleService {

  constructor(private httpBaseService: HttpBaseService) { }

  getAllRolesFullInfo(): Observable<APIResponseModel> {
    return this.httpBaseService.get(environment.httpBaseUrl + 'role/all');
  }

  getAllRolesBasicInfo(): Observable<APIResponseModel>{
    return this.httpBaseService.get(environment.httpBaseUrl + 'role/all-basic');
  }

  deleteRole(roleName: string): Observable<APIResponseModel> {
    return this.httpBaseService.delete(environment.httpBaseUrl + 'role/' + roleName);
  }

  updateRole(role: Role): Observable<APIResponseModel> {
    return this.httpBaseService.put(environment.httpBaseUrl + 'role', role);
  }
}

File: Backend/src/main/java/com/vcb/backend/service/RoleService.java
package com.vcb.backend.service;

import com.vcb.backend.dto.RoleDto;

import java.util.List;

public interface RoleService {
  public List<RoleDto> getAllRolesFullInfo();

  public List<RoleDto> getAllRolesBasicInfo();

  public RoleDto createRole(RoleDto roleDto);

  public RoleDto updateRole(RoleDto roleDto);

  public void deleteRole(String RoleName);
}
